### PR TITLE
Move abi-spec to status-go

### LINKED ISSUE #13432: Move abi-spec to status-go
Move abi-spec to status-go

#### `sign-transaction-button-clicked-from-chat`

```clojure
:data     (abi-spec/encode "transfer(address,uint256)" [to-norm amount-hex])
```

#### `sign-transaction-button-clicked-from-request`

```clojure
(abi-spec/encode "transfer(address,uint256)" [to-norm amount-hex])
```

#### `sign-transaction-button-clicked`

```clojure
(abi-spec/encode "transfer(address,uint256)" [to-norm amount-hex])
```

####  `get-transfer-token`

```clojure
(abi-spec/decode
   (str "0x" (subs data 10))
   (if (= type :approve-and-call) ["address" "uint256" "bytes"] ["address" "uint256"]))
```

#### `hex-to-number`

```clojure
(abi-spec/hex-to-number %)
```

#### `number-to-hex`

```clojure
(abi-spec/number-to-hex %)
```

### PR DESCRIPTION
**Pull request by qfrank**
fixes #13432

### CODE REVIEW

File: src/status_im/utils/varint.cljs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 (ns ^{:doc "Implementation of varint based on https://github.com/chrisdickinson/varint"}
  status-im.utils.varint
-  (:require [status-im.ethereum.abi-spec :as abi-spec]
-            ["web3-utils" :as utils]))
+  (:require ["web3-utils" :as utils]
```
Comment (flexsurfer) [inline src/status_im/utils/varint.cljs]: hope we can remove `web3-utils` dependency now :)

**Comment by flexsurfer** [comment]
@qfrank awesome job ! are you interested to try using JSI for these  new methods ? as an extra task but in this PR

File: src/status_im/utils/varint.cljs (hunk 1 above)
Comment (qfrank) [reply src/status_im/utils/varint.cljs]: Hi @flexsurfer ,  i found we used following functions with web3-utils in other places:
- `hexToUtf8`
- `sha3`
- `utf8ToHex`
- `isAddress`
- `toChecksumAddress`
- `checkAddressChecksum`

Can we make a separate issue for removing dependency `web3-utils` ?

Comment (flexsurfer) [reply src/status_im/utils/varint.cljs]: yes would be great

**Comment by flexsurfer** [comment]
@qfrank tests still fail, btw now you should have access to the logs

**Comment by qfrank** [comment]
> @qfrank tests still fail, btw now you should have access to the logs

Sorry for the "noise", actually i knew the tests still fail, when it's ready to review again, i'll re-request review to notify u, now it's ready! :) @flexsurfer

File: src/status_im/utils/varint.cljs (hunk 1 above)
Comment (qfrank) [reply src/status_im/utils/varint.cljs]: could u create the issue? i'll take it then, thanks @flexsurfer 

> yes would be great

File: src/status_im/signing/core.cljs [hunk 2]
```diff
@@ -535,7 +534,7 @@
                                  {:to    to-norm
                                   :value amount-hex}
                                  {:to   (ethereum/normalized-hex address)
-                                  :data (abi-spec/encode
+                                  :data (status/encodeFunctionCall
```
Comment (flexsurfer) [inline src/status_im/signing/core.cljs]: just wondering, could it be something like `(status/encodeTransfer [to-norm amount-hex])` and go-ethereum with sol file would be used ?

Comment (flexsurfer) [reply src/status_im/signing/core.cljs]: same question for decodeParameters

Comment (qfrank) [reply src/status_im/signing/core.cljs:537]: sorry, i'm not sure I clearly understand your question :) 
- do u mean u don't like the name `encodeFunctionCall`, and want to replace it to `encodeTransfer`?  actually the names i refferenced is [here](https://web3js.readthedocs.io/en/v1.2.7/web3-eth-abi.html#encodefunctioncall)
- go-ethereum don't need to use sol file, u can c the corresponding PR on the side of `status-go` 
@flexsurfer

File: src/status_im/utils/varint.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/utils/varint.cljs]: https://github.com/status-im/status-mobile/issues/13888

**Review by flexsurfer — APPROVED** [review]
great job!

**Comment by flexsurfer** [comment]
QA: please test tokens transactions

**Comment by flexsurfer** [comment]
@qfrank could you pls rebase and resolve conflicts , thanks

**Comment by qfrank** [comment]
> @qfrank could you pls rebase and resolve conflicts , thanks

@flexsurfer @churik done :)

**Comment by pavloburykh** [comment]
@qfrank thanx for the PR. Please, take a look at following issues

### ISSUE 1 Wallet balance is not being updated after refreshing wallet screen

**Preconditions:** recover some account having tokens on its balance

**Steps:** 
1. After account recovery go to wallet tab
2. Refresh the page in order to update tokens' balance
3. See if balance is updated

**Actual result:** balance has not been updated. It is updated after re-login or using "Scan tokens" option in wallet menu. 


https://user-images.githubusercontent.com/97245802/188659876-709ffa82-d2bf-4bdf-81fa-803c63c49709.mp4

[geth.log](https://github.com/status-im/status-mobile/files/9497890/geth.log)


**Expected result:** wallet balance has been updated 

### ISSUE 2 Null is not an object error on opening transaction from history

**Steps:**
1. Generate some transaction or recover an account that has some transactions in history
2. Open wallet tab
3. Open an account that has history of transactions
4. Tap history tab
5. Open any of transactions from the list
6. Observe an error

**Actual result:** for IOS 'null is not an object...' error. for Android 'Cannot read property 'times' of null' error. 

photo_2022-09-06 18 09 46
photo_2022-09-06 18 09 49

https://user-images.githubusercontent.com/97245802/188671021-5ad634d6-d51c-4134-9a8b-f18b696d7bf8.mp4

**Expected result:** no error. Transaction details are opened. 


### ISSUE 3 Newly sent transaction disappears from history after refresh

**Steps:**
 
1. Recover multiaccount that has got some tokens 
2. Open wallet tab
3. Sent some tokens to other account to generate transaction in history list
4. Open history tab
5. Refresh the page and see if sent transaction is still in the list

**Actual result:** sent transaction disappears from history

https://user-images.githubusercontent.com/97245802/188675753-f91551a4-72dd-43f5-ac2f-2363db238949.mp4

[geth.log](https://github.com/status-im/status-mobile/files/9498462/geth.log)

**Expected result:** sent transaction remains in history list

**Comment by pavloburykh** [comment]
### ISSUE 4 Sticker market is not loaded

Works fine on latest nightly. Fail to load on current PR builds. 

**Steps:** 
1. Open 1-1 chat
2. Tap sticker icon
3. Tap Get Stickers to open market
4. See if Sticker items are loaded

**Actual result:** endless spinner, stickers fail to load. 

https://user-images.githubusercontent.com/97245802/188678992-4df833bc-802d-43b7-8ba4-e38f0de42b07.mp4

[geth.log](https://github.com/status-im/status-mobile/files/9498564/geth.log)

**Expected result:** Stickerpacks list is loaded in stickermarket.

**Comment by qfrank** [comment]
Hi @pavloburykh , thanks for reporting, just fixed issue 1-3.
I think issue 4 is not relate to this PR, i tried switch status-mobile to commit id `d4b7e5c8b60fe0943ba116c5634995d4cd45574e` then executed `scripts/update-status-go.sh 26b33aa09db0b19fd7fc94493531f120b05f6fb4`, it has the same issue

**Comment by flexsurfer** [comment]
hey @qfrank, stickers bug mentioned is in the status-go, you need to rebase your status-go to the latest version and then update version in this PR , thank you

**Comment by qfrank** [comment]
> hey @qfrank, stickers bug mentioned is in the status-go, you need to rebase your status-go to the latest version and then update version in this PR , thank you

@flexsurfer @pavloburykh done :)

**Comment by pavloburykh** [comment]
@qfrank could you please also take a look at integration test builds which are failing https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-13858/6/consoleText
Move abi-spec to status-go by qfrank · Pull Request #13858 · status-im:status-mobile 2022-09-07 11-26-51

**Comment by pavloburykh** [comment]
according to current flow we should make sure that integration tests are passed before taking PR into test.

**Comment by qfrank** [comment]
Hi @pavloburykh , could u pls post the log files here, i've no permission to access jenkins ): thanks

**Comment by flexsurfer** [comment]
@qfrank its lint
```
[2022-09-07T08:23:16.342Z] src/status_im/ethereum/decode.cljs:2:14: warning: namespace status-im.native-module.core is required but never used

[2022-09-07T08:23:16.342Z] linting took 4610ms, errors: 0, warnings: 1
```

**Comment by flexsurfer** [comment]
@pavloburykh if its lint you can take PR for testing

**Comment by qfrank** [comment]
lint fixed :)

**Comment by pavloburykh** [comment]
@qfrank hi! I see that you pushed some more commits. Are those with some additional fixes? Sorry, I haven't finished yet with testing current PR as I had to switch to testing release. Will continue after release testing.

**Comment by qfrank** [comment]
> @qfrank hi! I see that you pushed some more commits. Are those with some additional fixes? Sorry, I haven't finished yet with testing current PR as I had to switch to testing release. Will continue after release testing.

Hi @pavloburykh, it's not additional fixes, just want to make it ready to merge 
image

**Comment by pavloburykh** [comment]
@qfrank please, take a look at issue 5

### ISSUE 5 Some of app elements (e.g Profile menu items, trans history list items etc) periodically stop responding on Android

Currently there are no specific steps to reproduce the bug. I have faced it several times already on Android build downloaded from this PR. I haven't faced this bug on nightly or other PRs so I assume it might be related to this PR. 

The last time I faced the bug my **steps were the following:** 

1. Restored the account that has balance and some history of transactions
2. Right after restore went to Wallet tab - opened history tab - waited until history was loaded
3. Opened details of couple of transactions
4. Went back to account page
5. Opened send transaction screen, opened receive transaction screen
6. Backgrounded app
7. Reopened app
8. Went to profile tab
9. Opened Add contact screen
10. Closed add contact screen
11. Tried to enter some other Profile pages

**Actual result:** Profile page menu items stopped responding as well as many others. 

Repeating the above steps does not guarantee reproduction. I Attach log file. Hope it will help to find out a rootcause. 


https://user-images.githubusercontent.com/97245802/189622440-a451d324-e38d-47a4-a73a-dee12e0bc6e5.mp4

[geth.log](https://github.com/status-im/status-mobile/files/9546773/geth.log)

@qfrank please, ping me if you need any other help in debugging. 

**OS:** Android 11
**Device:** Samsung Galaxy A52

**Comment by flexsurfer** [comment]
seems like something wrong with the RN bridge

**Comment by pavloburykh** [comment]
@qfrank also, please take a look at IOS builds. They are currently failing due to conflicts in status go 
https:::ci status im:job:status-mobile:job:prs:job

**Comment by qfrank** [comment]
Hi @pavloburykh , sorry for the delay, it was mid autumn festival, just rebased with latest branch `develop` , i'll  take a look at issue 5 later

**Comment by qfrank** [comment]
image the latest rebase has issues as screenshot shows, i'll continue after it get resolved on `develop`

**Comment by qfrank** [comment]
Hi @pavloburykh , i tried your steps but has no luck to reproduce it, i didn't found useful info from geth.log, can u still reproduce issue 5 with latest rebasing?

**Comment by flexsurfer** [comment]
@qfrank pls fix the lint

**Comment by pavloburykh** [comment]
@qfrank thank you for the PR. I have not reproduced ISSUE 5 after rebase, will hope that it is fixed. Ready for merge.

**Comment by flexsurfer** [comment]
@qfrank could you resolve version, and we'll merge it

**Comment by qfrank** [comment]
sure, just saw your message :) will do it when u r back on monday

**Pull request merged**